packages/transaction-controller/src/helpers/PendingTransactionTracker.ts

```diff
@@ -40,12 +45,22 @@ export class PendingTransactionTracker {
     this.#getChainId = getChainId;
     this.#getEthQuery = getEthQuery;
     this.#getTransactions = getTransactions;
+    this.#nonceTracker = nonceTracker;
   }
 
   start() {
     // eslint-disable-next-line @typescript-eslint/no-misused-promises
     this.#blockTracker.addListener('latest', async () => {
-      await safelyExecute(() => this.#onLatestBlock());
+      const nonceGlobalLock = await this.#nonceTracker.getGlobalLock();
```

Comment:
❤️ Thanks for adding this in too!

It was something I noticed in the extension that seems wise in case we have rapid blocks and slow queries and thus potential race conditions.